service/gov_pay.go

```diff
@@ -118,18 +112,6 @@ func (gp *GovPayService) CreatePaymentAndGenerateNextURL(req *http.Request, paym
 	return govPayResponse.GovPayLinks.NextURL.HREF, Success, nil
 }
 
-// To get the status of a GovPay payment, GET the payment resource from GovPay and return the State block
-func (gp *GovPayService) getGovPayPaymentState(paymentResource *models.PaymentResourceRest, cfg *config.Config) (*models.State, ResponseType, error) {
-
-	govPayResponse, err := callGovPay(gp, paymentResource)
-	if err != nil {
-		return nil, Error, err
-	}
-
-	// Return the status of the payment
-	return &govPayResponse.State, Success, nil
-}
-
 // GetPaymentDetails gets the details of a GovPay payment
 func (gp *GovPayService) GetPaymentDetails(paymentResource *models.PaymentResourceRest) (*models.PaymentDetails, ResponseType, error) {
```

Comment:
Would we be better off returning the `provider_id` in this function and endpoint rather than the Provider Status? This is normally called to retrieve information about a payment on GovPay for things such as card type by other services. Should just be a case of adding a new field to the `models.PaymentResourceRest` model